src/file.coffee

```diff
@@ -130,13 +128,29 @@ class File
 
   # Public: Returns a {Boolean}, true if the file exists, false otherwise.
   exists: ->
+    Q.Promise (resolve, reject) =>
+      fs.exists @getPath(), resolve
+
+  # Public: Returns a {Boolean}, true if the file exists, false otherwise.
+  existsSync: ->
     fs.existsSync(@getPath())
 
   # Public: Get the SHA-1 digest of this file
   #
-  # Returns a {String}.
+  # Returns a promise that resolves to a {String}.
   getDigest: ->
-    @digest ? @setDigest(@readSync())
+    return Q(@digest) if @digest
+    @read().then (contents) =>
+      # read sets digest
+      @digest
+
+  # Public: Get the SHA-1 digest of this file
+  #
+  # Returns a {String}.
+  getDigestSync: ->
+    contents = @readSync()
```

Comment:
I'd say just ditch the `contents` variable if it is never used and make this line just `@readSync()`